File: Client/HoneyClient/src/app/data/interface/product/Product.ts
import { ProductStatusEnum } from "../../enum/ProductStatusEnum";

export interface ProductItem {
    id: number; // Идентификатор продукта
    name: string; // Название продукта
    description: string; // Описание продукта
    price: number; // Цена продукта
    status: string; // Статус продукта (например, Available, Out of Stock)
    files: number[]; // Список идентификаторов файлов
  }
  export interface CreateProduct {
    name: string;
    description?: string;
    price: number;
    status: ProductStatusEnum;
    files: number[];
  }
  
  export interface UpdateProduct {
    id: number;
    name: string;
    description?: string;
    price: number;
    statusId: number;
    files: number[];
  }

File: Backend/HoneyShop.Domain/Model/Product.cs
namespace HonaeyShop.Domain.Model;

// Модель продукта
public class Product
{
    public int Id { get; set; } // Идентификатор продукта
    public string Name { get; set; } = string.Empty;// Название продукта
    public string Description { get; set; } = string.Empty;// Описание продукта
    public decimal Price { get; set; } // Цена продукта
    
    public int StatusId { get; set; } // Идентификатор статуса продукта
    public ProductStatus  Status{ get; set; }  // Статус продукта 
    public DateTime CreatedAt { get; set; } // Дата создания записи
    public DateTime UpdatedAt { get; set; } // Дата обновления записи
    
    public bool IsRemoved { get; set; }
    public ICollection<Cart> Carts { get; set; } // Связь с корзинами
    
    public ICollection<OrderDetail> OrderDetails { get; set; } // Связь с деталями заказов
    
    public ICollection<ApplicationFile> Files { get; set; } // Связь с файлами
    
    public ICollection<History> Histories { get; set; } // Связь с историей изменений
}